articles/quickstart/webapp/nodejs/01-login.md

```diff
@@ -14,4 +16,239 @@ github:
+:::
+
+Create a new router `routes/auth.js`:
+
+```js
+// routes/auth.js
+
+var express = require('express');
+var router = express.Router();
+var Auth0Strategy = require('passport-auth0'),
+    passport = require('passport');
+var ensureLoggedIn = require('connect-ensure-login').ensureLoggedIn();
+
+// Perform the login, after login Auth0 will redirect to callback
+router.get('/login',
+  passport.authenticate('auth0', {scope: 'openid email profile'}), function (req, res) {
+  res.redirect("/");
+});
+
+// Perform the final stage of authentication and redirect to '/user'
+router.get('/callback',
+  passport.authenticate('auth0', { failureRedirect: '/login' }),
+  function(req, res) {
+    if (!req.user) {
+      throw new Error('user null');
+    }
+    res.redirect("/user");
+  }
+);
+
+/* GET user profile. */
+router.get('/user', ensureLoggedIn, function(req, res, next) {
+  res.render('user', {
+    user: req.user ,
+    userProfile: JSON.stringify(req.user, null, '  ')
+  });
+});
+
+// Perform session logout and redirect to homepage
+router.get('/logout', (req, res) => {
+  req.logout();
+  res.redirect('/');
+});
+
+module.exports = router;
+```
+
+Include this new router in your app, by requiring the file in the top section of  `app.js`:
+
+```js
+// app.js
+
+var indexRouter = require('./routes/index');
+var authRouter = require('./routes/auth');
+
+//..
+//..
+
+app.use('/', indexRouter);
+//..
+app.use('/', authRouter);
+//..
+```
+
+![login page](/media/articles/web/hosted-login.png)
+
+### Implement a Helper Function to Check User Session
+
+In the views and layouts, we will need to conditionally render content depending on if a user is logged in or not.
+Add a helper function to check if the user is persisted in the session or not.
+
+```js
+// app.js
+
+// Look up session to know if user is logged in 
+app.use(function(req, res, next) {
+  res.locals.loggedIn = false;
+  if (req.session.passport && typeof req.session.passport.user != 'undefined') {
+    res.locals.loggedIn = true;
+  }
+  next();
+});
+
+// the auth router should be loaded after the function definition
+app.use('/', indexRouter);
+//..
+app.use('/', authRouter);
+//..
+```
+
+### Implement Navigation Links 
+
+In your views, use the function defined in the previous step to render links to login, logout an user profile, depending on the case.
+
+```pug
+// views/layout.pug
+
+  body
+    // ...
+    a(href="/") Home
+    if loggedIn
+      a(href="/user") Profile
+      a(href="/logout") Logout
+    else
+      a(href="/login") Login
+    
+    // ...
+    block content
+```
+
+## Display User Information
+
+Modify the `/user` endpoint to display the user object. Update the `views/user.pug` template.
+
+```pug
+// views/user.pug
+
+extends layout
+
+block content
+  br
+  img(src=user.picture)
+  p Welcome #{user.nickname}!
+  br
+  h2 User Profile
+  pre #{userProfile}
+```
```

Comment:
Maybe we need to wrap up the document saying "if your un the app and navigate to .." or "you log in', you'll be able to see the user profile after logging in.